Assignment5.txt

```diff
@@ -0,0 +1,15 @@
+Clarke Littlejohn
+
+1) calling that function would look something like
+	fxn(x,y,z);
+	this is assuming that x y and z are all int data types adn have been initialized and everything else is declared;
+	You dont need to put an int in the parameter because that would be like put int a = int x; which is wrong you would
+	put int a = x; which is what the parameter is doing.
+
+2) iteration is a loop for/while statement. one iteration is one loop through, recursion is calling a function x in function x to get a loop.
+   recurison is iteration for functions. recursion is sometimes better than iteration because it is makes for more organized code, it can be
+   easier to code a loop in recursion, and it can better more efficent. Recursion is worse in most cases as it takes up much for memory than iteration.
+
+3)A compiler translates one code into another. From what i understand and from the slides is that it turns the source code into some lower level language
```

Comment:
In short, a compiler converts a program from a human-readable format into a machine-readable format. 
Lexing - break up the text of the program into "tokens" 
Parsing - convert the sequence of tokens into a parse tree, which is a data structure representing various language constructs: type declarations, variable declarations, function definitions, loops, conditionals, expressions, etc. 
Optimization - evaluate constant expressions, optimize away unused variables or unreachable code, unroll loops if possible, etc. 
Translate the parse tree into machine instructions (or JVM byte code)